client/pong.c

```diff
@@ -124,6 +202,28 @@ static int process_server_message(Pong* pong, ServerMessage* message) {
       pong->game_session.state = WAITING_FOR_LOBBY;
       LOG_INFO("Player with IP: %s has joined your session", pong->game_session.opponent_ip);
       break;
+
+    case SERVER_UPDATE:
+      if (pong->game_session.state != PLAYING) {
+        LOG_INFO("Got first server update, change game_session.state to playing");
+        pong->game_session.state = PLAYING;
+      }
+
+      pong->game.opponent.position.x = message->server_update.opponent_position.x;
+      pong->game.opponent.position.y = message->server_update.opponent_position.y;
+      pong->game.ball.position.x = message->server_update.ball_position.x;
+      pong->game.ball.position.y = message->server_update.ball_position.y;
+
+      prepare_client_message(pong);
```

Comment:
This logic introduces unnecessary delay between updates. We shouldn't wait for server update to send client update. Also, the name of this function is misleading.

Reply:
`prepare_client_message` is also called in `pong_process_network`. Why is it also called there?